[ui][fix] Application Menubar: Ensure Menubar occupies all of available height of the layout

waaake opened the pull request:
## Description
This PR modifies the `Menubar` component to ensure it fills the layout in terms of height.
The change addresses an inconsistency where the MenuBar did not align with the Parent's dimensions.

## Details of the change
* Parenting the Menubar inside the Rectangle matching the colour of the background header ensures the additional area is filled correctly.
* Aligning the Menubar at the bottom, leaving a pixel to match the placement of the HomeButton.
* HomeButton and ExecutionButtons have reduced padding to reduce the overall area of the button while retaining the icon size.

The above changes ensure the overall alignment of the Header bar elements, Custom buttons and MenuBar.

## Appearence after the changes
![image](https://github.com/user-attachments/assets/0f5f46bb-29ea-46ac-aefc-45ac8f916b1c)

cbentejac commented:
The main issue I have with this change is that, although the menu bar now takes up the whole space as it should, the menus themselves are not vertically centered with the homepage icon. The menu bar starts slightly higher than the homepage button but finishes slightly higher as well, and it's very clear when hovering.

It'd be great if the button and the menu bar could be aligned.

fabiencastan commented on meshroom/ui/qml/Application.qml:
```diff
@@ -624,320 +626,327 @@ Page {
+            MenuBar {
+                palette.window: Qt.darker(activePalette.window, 1.15)
+                anchors.bottom: parent.bottom
+                anchors.bottomMargin: 1     /// this margin counters the offset vertically when lining up against with the homeButton
```
Looks better to me without this offset.
```suggestion
```

fabiencastan commented on meshroom/ui/qml/Application.qml:
```diff
@@ -624,320 +626,327 @@ Page {
+
+            MenuBar {
+                palette.window: Qt.darker(activePalette.window, 1.15)
+                anchors.bottom: parent.bottom
```
if we could avoid anchors that's ideal.

fabiencastan commented on meshroom/ui/qml/Application.qml:
```diff
@@ -624,320 +626,327 @@ Page {
-                        _reconstruction.new()
-                    })
-                }
+        Rectangle {
+            /**
+             * This Rectangle serves as the parent of the menubar
+             * occupying all of the additional space which the MenuBar is not going to get
+             */
+            Layout.fillWidth: true
+            Layout.fillHeight: true
+            color: Qt.darker(activePalette.window, 1.15)
+
```
Just using "height: parent.height" on the MenuBar works for me, without adding a Rectangle into the hierarchy.

waaake replied:
This rectangle already existed to cover up additional space horizontally, but I could try using height property to set this, last I can recall, there was a very minor pixel worth of offset against the Custom buttons. Let me try the changes, thanks.

waaake committed:
[ui] Application: Fixing the height of the Menubar to align with neighbours

Added a base rectangle serving as the header for the Application. The Child Buttons now get a icon color change on hover

Relaced Rectangle Spacer with Item to get the color from background rect

Fixed the height of the Menubar to occupy available height
